### ISSUE TITLE
Warn when private structs have public fields

### ISSUE THREAD

**Issue by mahkoh** [rust-lang/rust#17697]
```rust
#![crate_type = "lib"]

struct X {
    pub i: int,
}

pub fn f() {
    X { i: 1 };
}
```

**Comment by frewsxcv** [comment]
We still don't warn with the latest Rust. This issue is still valid.
